src/backend/clearboard/main.py

```diff
@@ -47,3 +96,124 @@ async def post_picture(file: UploadFile = File(...)):
         with open(f"{path}/{file.filename}", 'wb') as f:
             shutil.copyfileobj(file.file, f)
         return {"message": file.filename}
+
+
+def image_to_base64(img: np.ndarray) -> bytes:
+    """ Given a numpy 2D array, returns a JPEG image in base64 format """
+    img_buffer = cv2.imencode('.jpg', img)[1]
+    return base64.b64encode(img_buffer).decode('utf-8')
+
+
+def get_image(image):
+    return image_to_base64(image)
+
+
+@app.get("/process")
+async def photo(roomName: str, process: str):
+    try:
+        img_cropped_path = './' + roomName + '/' + roomName + 'cropped.jpg'
+        original_img_path = './' + roomName + '/' + roomName + '.jpg'
+        processed_img_path = './' + roomName + '/' + roomName + process + ".jpg"
+        if os.path.exists(os.path.abspath(img_cropped_path)):
+            img_to_process = img_cropped_path
+        elif os.path.exists(os.path.abspath(original_img_path)):
+            img_to_process = original_img_path
+
+        if process == 'Color':
+            color.whiteboard_enhance(cv2.imread(
+                img_to_process), processed_img_path)
+        elif process == 'B&W':
+            blackNwhite.black_n_white(img_to_process, processed_img_path)
+        elif process == 'Contrast':
+            contrast.enhance_contrast(img_to_process, processed_img_path)
+        elif process == 'original':
+            processed_img_path = img_to_process
+        elif process == 'SuperRes':
+            super_res.super_res(img_to_process, processed_img_path, "./clearboard/EDSR/EDSR_x4.pb")
+        else:
+            processed_img_path = img_to_process
+        img = cv2.imread(processed_img_path)
+        volume = np.asarray(img)
+        image = get_image(volume)
+        print('image sent')
+        return Response(content=image)
+
+    except:
+        print('no file to send')
+
+
+@app.get("/original_photo")
+async def photo(roomName: Optional[str] = None):
+    try:
+        original_img_path = './' + roomName + '/' + roomName + '.jpg'
+        if os.path.exists(os.path.abspath(original_img_path)):
+            img = cv2.imread(original_img_path)
+            volume = np.asarray(img)
+            image = get_image(volume)
+            print('image sent')
+            return Response(content=image)
+        else:
+            print('file not found')
+            time.sleep(1)
+    except:
+        print('no file to send')
+
+
+@app.websocket("/ws1")
+async def websocket_endpoint(websocket: WebSocket):
+    await manager.connect(websocket)
+    temps = 0
+    temps_coord = 0
+    try:
+        while True:
+            print('test')
+            roomName = await websocket.receive_text()
+            original_img_path = './' + roomName + '/' + roomName + '.jpg'
+            img_cropped_path = './' + roomName + '/' + roomName + 'cropped.jpg'
+
+            if (os.path.isfile(os.path.abspath(original_img_path)) and (temps != os.path.getctime(original_img_path))):
```

Comment:
can you add comments here to explain what are these cases? In general, your code is lacking comments whenever you do something that is not self explanatory